src/libs/decoders/opus.cpp

```diff
@@ -137,12 +137,13 @@ static int32_t RWops_opus_seek(void * stream, const opus_int64 offset, const int
 {
     // Guard against invalid inputs
     assertm(stream, "OPUS: Input is not initialized");
+    assertm(offset <= INT32_MAX, "OPUS: requested offset beyond 2 GiB limit");
     assertm(whence == SEEK_SET || whence == SEEK_CUR || whence == SEEK_END,
             "OPUS: The position from where to seek is invalid");
 
-    const int64_t offset_after_seek = SDL_RWseek(static_cast<SDL_RWops*>(stream),
-                                                 static_cast<int32_t>(offset),
-                                                 whence);
+    const auto offset_after_seek = SDL_RWseek(static_cast<SDL_RWops*>(stream),
+                                              static_cast<int32_t>(offset),
```

Comment:
This static cast is unnecessary for SDL2 (it was likely needed for SDL1). Does it mean assert is not needed either?

Using auto is not needed either in this case - SDL2 returns Sint64 here, which is the same type as int64_t, so the code was good already (except %ld in SNDDBG below, but using auto does not solve that).

Reply:
`offset` is a long long int at least Ubuntu 18.04:

`opus.cpp:148:38: warning: format '%ld' expects argument of type 'long int', but argument 2 has type 'long long int' [-Wformat=]`

So needed to retain one cast down to `int64_t`:

``` patch
     SNDDBG(("Opus ops seek:          "
-            "requested: %" PRId64 " and got: %" PRId64"\n",
-            offset, offset_after_seek));
+            "requested: %" PRId64 " and got: %" PRId64 "\n",
+            static_cast<int64_t>(offset), offset_after_seek));
```